### PR TITLE
Acceptance test to filter resources using keyword in the favourites page

### LINKED ISSUE #2709: Add acceptance test to filter files, folders, hidden elements using keyword in the favourites page
Related to issue #2482
 - [x] -Add new tests to filter files, folders and hidden elements using keyword in the favourites page

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Acceptance test to filter files, folders and hidden elements using keyword in the favourites page

## Related Issue

- Fixes #2709 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/helpers/webdavHelper.js [hunk 1]
```diff
@@ -232,3 +232,30 @@ exports.uploadFileWithContent = function (user, content, filename) {
...
+      headers,
+      body
+    })
+    .then(res => res.text())
+    .then(res => {
+      const favData = convert.xml2js(res, { compact: true })['d:multistatus']['d:response']
+      const favItems = []
+      favData.map(favourite => {
+        favItems.push({
+          href: favourite['d:href']._text,
```
Comment (dpakach) [inline tests/acceptance/helpers/webdavHelper.js]: Do we really need the href here. Can't we just parse out the name of file or folder.

File: tests/acceptance/helpers/webdavHelper.js [hunk 2]
```diff
@@ -232,3 +232,30 @@ exports.uploadFileWithContent = function (user, content, filename) {
...
     .then(res => httpHelper.checkStatus(res, 'Could not upload file' + filename + 'with content' + content))
 }
+
+exports.getFavouritedResources = function (user) {
+  const headers = httpHelper.createAuthHeader(user)
+  const body = `<oc:filter-files  xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns"><d:prop><d:resourcetype /></d:prop>
+         <oc:filter-rules>
+        <oc:favorite>1</oc:favorite>
+    </oc:filter-rules>
+</oc:filter-files>`
```
Comment (dpakach) [inline tests/acceptance/helpers/webdavHelper.js]: Format this properly

File: tests/acceptance/helpers/webdavHelper.js (hunk 1 above)
Comment (jasson99) [reply tests/acceptance/helpers/webdavHelper.js]: fixed

File: tests/acceptance/helpers/webdavHelper.js (hunk 2 above)
Comment (jasson99) [reply tests/acceptance/helpers/webdavHelper.js]: done

File: tests/acceptance/helpers/filesFoldersHelper.js [hunk 3]
```diff
@@ -78,3 +78,15 @@ exports.getSharedWithOthersFiles = async function (user) {
   return sharedWithOthersElements.filter(elements => elements.mimetype.includes('application/octet-stream'))
     .map(elements => normalize(elements.path))
 }
+
+exports.getFavouritedFolders = async function (user) {
+  const favElems = await getFavouritedResources(user)
+  const favFolders = favElems.filter(element => element.isFolder).map(element => filename(element.resource))
```
Comment (dpakach) [inline tests/acceptance/helpers/filesFoldersHelper.js]:
```suggestion
  const favFolders = favElems.filter(element => element.isFolder)
```

File: tests/acceptance/helpers/filesFoldersHelper.js [hunk 4]
```diff
@@ -78,3 +78,15 @@ exports.getSharedWithOthersFiles = async function (user) {
...
+  return favFolders
+}
+
+exports.getFavouritedFiles = async function (user) {
+  const favElems = await getFavouritedResources(user)
+  const favFiles = favElems.filter(element => !element.isFolder).map(element => filename(element.resource))
```
Comment (dpakach) [inline tests/acceptance/helpers/filesFoldersHelper.js]:
```suggestion
  const favFiles = favElems.filter(element => !element.isFolder)
```

File: tests/acceptance/helpers/filesFoldersHelper.js (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/helpers/filesFoldersHelper.js]: fixed

File: tests/acceptance/helpers/filesFoldersHelper.js (hunk 4 above)
Comment (jasson99) [reply tests/acceptance/helpers/filesFoldersHelper.js]: fixed

File: tests/acceptance/helpers/filesFoldersHelper.js (hunk 3 above)
Comment (jasson99) [reply tests/acceptance/helpers/filesFoldersHelper.js]: the mapping is required anyways

File: tests/acceptance/helpers/filesFoldersHelper.js (hunk 4 above)
Comment (jasson99) [reply tests/acceptance/helpers/filesFoldersHelper.js]: mapping is needed

**Pull request merged**